receiver/splunkenterprisereceiver/scraper.go

```diff
@@ -56,13 +56,22 @@ func (s *splunkScraper) start(ctx context.Context, h component.Host) (err error)
 // listens to the error channel and combines errors sent from different metric scrape functions,
 // returning the combined error list should context timeout or a nil error value is sent in the
 // channel signifying the end of a scrape cycle
-func errorListener(eQueue <-chan error, eOut chan<- *scrapererror.ScrapeErrors) {
+func errorListener(ctx context.Context, eQueue <-chan error, eOut chan<- *scrapererror.ScrapeErrors) {
 	errs := &scrapererror.ScrapeErrors{}
 
-	for err := range eQueue {
-		errs.Add(err)
+	for {
```

Comment:
Should this be considered a bug fix, since it appears that this was originally leaking goroutines (memory)?

Reply:
I actually don't think it was ever leaking since the enclosing scrape function did call `close()` on the channel (which would end the `for err := range eQueue` loop, I think). That said I could be wrong and whether or not we consider it a bug fix or a chore makes no real difference to me!

Reply:
Thanks for clarifying, you're right, since the channel was already being closed it wouldn't be a leak. 👍